R/aux_functions.R

```diff
@@ -45,7 +48,11 @@ inf_fn <- function(inc_samp = NULL, k = NULL) {
 #'   quarantine = FALSE
 #' )
 #' extinct_prob(res, cap_cases = 4500)
-extinct_prob <- function(outbreak_df_week = NULL, cap_cases  = NULL, week_range = 12:16) {
+extinct_prob <- function(outbreak_df_week, cap_cases, week_range = 12:16) {
+
+  checkmate::assert_data_frame(outbreak_df_week)
+  checkmate::assert_number(cap_cases, lower = 0)
+  checkmate::assert_numeric(week_range)
```

Comment:
since this always routes thru detect_extinct, perhaps just catch there? The mild complexity associated with getting the error entry point right would be worth it, imo.

i'd start this function immediately calculating extinct_runs, then do the other steps.

Reply:
also, is week range numeric or an integer?

Reply:
It's an `integer`, assert updated in 9740413ab8a1c13504b32ef42ffe1052c8fe468e. 

On the first point I'll leave as is until we refactor the package API and decide what to do with `extinct_prob()` and `detect_extinct()`.

Reply:
This commit is on the feature branch of the fork that is being merged in this PR, therefore merging this PR will include this commit.